Assets/Plugins/iOS/Shopify/WebCheckoutViewController.swift

```diff
@@ -0,0 +1,345 @@
+//
+//  WebCheckoutViewController.swift
+//  UnityBuySDK
+//
+//  Created by Shopify.
+//  Copyright © 2017 Shopify Inc. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit
+import WebKit
+
+// MARK: View Controller Logic
+class WebCheckoutViewController: UIViewController {
+    weak var delegate: WebCheckoutDelegate?
+    
+    fileprivate(set) var url: URL
+    
+    fileprivate var checkoutView: WebCheckoutView {
+        return view as! WebCheckoutView
+    }
+    
+    fileprivate var webView: WKWebView {
+        return checkoutView.webView
+    }
+    
+    init(url: URL, delegate: WebCheckoutDelegate) {
+        self.url = url
+        self.delegate = delegate
+        
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func loadView() {
+        self.view = WebCheckoutView(frame: UIScreen.main.bounds)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        webView.navigationDelegate = self
+        checkoutView.actionButton.addTarget(self, action: #selector(didPressCancel), for: .touchUpInside)
+        
+        loadCheckoutURL()
+    }
+}
+
+// MARK: Actions
+extension WebCheckoutViewController {
+    func loadCheckoutURL() {
+        let request = URLRequest(url: url)
+        webView.load(request)
+    }
+    
+    func didPressCancel() {
+        delegate?.willDismiss()
+        self.dismiss(animated: true)
```

Comment:
Do we want to wait for a response from `willDismiss`? The oder of execution for `self.dismiss` and `willDismiss` is not guaranteed, since `willDismiss()` is sent async